src/main/java/com/example/spring/controller/CategoryController.java

```diff
@@ -0,0 +1,73 @@
+package com.example.spring.controller;
+
+import com.example.spring.dto.book.BookDto;
+import com.example.spring.dto.category.CategoryDto;
+import com.example.spring.dto.category.CategoryRequestDto;
+import com.example.spring.service.book.BookService;
+import com.example.spring.service.category.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/categories")
+@RequiredArgsConstructor
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")

```

Comment:
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```
fix in all places and make sure your security works as expected

Reply:
It doesn`t work if its not like ROLE_ADMIN

Reply:
you can try to fix it